SConstruct

```diff
@@ -1293,6 +1294,11 @@ if env['python_package'] != 'none':
                 "Found {0} but {1} or newer is required".format(
                 python_version, python_min_version))
             warn_no_full_package = True
+        elif python_version >= LooseVersion("3.8"):
+            # Reset the minimum Cython version if the Python version is 3.8 or higher
+            # Due to internal changes in the CPython API, more recent versions of
+            # Cython are necessary to build for Python 3.8
+            cython_min_version = LooseVersion("0.29.12")
```

Comment:
I think it would be helpful for the comments here to be slightly more clear that there's nothing Cantera-specific about any of this: Only Cython >= 0.29.12 is compatible with Python 3.8, and we're just setting this dependency so that users get a more useful error message than the compilation errors that would otherwise result.